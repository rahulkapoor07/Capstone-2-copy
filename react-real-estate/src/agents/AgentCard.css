.AgentCard{
    width: 100%;
    background-color: #694E4E;
    color: white;
    border-radius: 2rem;
    margin-bottom: 3rem;
    cursor: pointer;
    padding: 1rem;
    box-shadow: 8px 8px 5px grey;
}

.AgentCard img {
    width: 100px;
    height: 100px;
    border-radius: 50%;
}
.details{
    margin-bottom: .4rem;
}

.description{
    text-align: center;
}

.zoom {
    transition: transform .1s;
    text-align: center;
}

.zoom:hover {
    transform: scale(1.05);
}

