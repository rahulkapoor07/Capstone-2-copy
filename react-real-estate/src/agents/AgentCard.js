import React, { useContext } from 'react';
import { Link,useHistory } from "react-router-dom";
import "./AgentCard.css"
import UserContext from '../useContext/UserContext';

function AgentCard({agent_name, agent_phones, id, agent_photo, agent_description, agent_email, agent_type
    ,agent_web_url,office_address,office_name,office_website}) {
    const history = useHistory();
    const {setSingleAgentData} = useContext(UserContext);
    const singleData = {agent_name, agent_phones, id, agent_photo, agent_description, agent_email, agent_type
        ,agent_web_url,office_address,office_name,office_website}
    

    const handleClick = (e)=>{
        e.preventDefault();
        setSingleAgentData({...singleData})
        history.push(`/agents/${id}`)
    }
        return (
            <div className='AgentCard zoom'>
                <Link to={`/agents/${id}`} 
                style={{"background":"#694E4E", "color":"white", "textDecoration":"none"}}
                onClick={handleClick}>
                {singleData && <div className='row'>
                <div className='col-md-6'>
                    <div className='row'>
                        <div className='col-xl-3 col-lg-3'>
                            <img src={agent_photo} alt="photo"/>
                        </div>
                        <div className='col-xl-8 col-lg-6'>
                            <p className='details'><strong>{agent_name}</strong></p>
                            <p className='details'>{agent_phones ? agent_phones[0].number : null}</p>
                            <p className='details'>{agent_email}</p>
                            <p className='details'>{office_address ? `${office_address}` : null}</p>
                        </div>
                    </div>
                </div>
                <div className='description col-md-5'>
                    <p>{agent_description ?`${agent_description.split("").splice(0,180).join("")}...` : `No Description`}</p>
                    </div>
              </div>}
                </Link>         
          </div>
    )
}

export default AgentCard;
