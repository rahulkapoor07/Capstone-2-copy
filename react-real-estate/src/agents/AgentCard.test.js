import { render } from '@testing-library/react';
import AgentCard from './AgentCard';
import {MemoryRouter} from "react-router-dom";


test("render without crashing", ()=>{
    render(<MemoryRouter><AgentCard/></MemoryRouter>);
});

test("render without crashing", ()=>{
    const {asFragment} = render(<MemoryRouter><AgentCard/></MemoryRouter>);
    expect(asFragment()).toMatchSnapshot();
});