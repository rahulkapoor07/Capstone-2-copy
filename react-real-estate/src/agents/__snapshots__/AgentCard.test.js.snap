// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render without crashing 1`] = `
<DocumentFragment>
  <div
    class="AgentCard zoom"
  >
    <a
      href="/agents/undefined"
      style="background: rgb(105, 78, 78); color: white; text-decoration: none;"
    >
      <div
        class="row"
      >
        <div
          class="col-md-6"
        >
          <div
            class="row"
          >
            <div
              class="col-xl-3 col-lg-3"
            >
              <img
                alt="photo"
              />
            </div>
            <div
              class="col-xl-8 col-lg-6"
            >
              <p
                class="details"
              >
                <strong />
              </p>
              <p
                class="details"
              />
              <p
                class="details"
              />
              <p
                class="details"
              />
            </div>
          </div>
        </div>
        <div
          class="description col-md-5"
        >
          <p>
            No Description
          </p>
        </div>
      </div>
    </a>
  </div>
</DocumentFragment>
`;
